src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -813,24 +1421,22 @@ const encryptSubmission: ControllerHandler<
     publicKey,
   )
 
-  const encryptedVersion = {
+  const encryptedBody = {
     attachments: encryptedAttachments,
     responses: filteredResponses.value as EncryptFormFieldResponse[],
     encryptedContent,
-    version: req.body.version,
+    version: requestBody.version,
   }
 
-  req.body = encryptedVersion
-  return next()
+  return okAsync(encryptedBody)
 }
 
 export const handleStorageSubmission = [
   CaptchaMiddleware.validateCaptchaParams,
   TurnstileMiddleware.validateTurnstileParams,
   ReceiverMiddleware.receiveStorageSubmission,
   EncryptSubmissionMiddleware.validateStorageSubmissionParams,
-  encryptSubmission,
-  submitEncryptModeForm,
+  submitStorageModeForm,
```

Comment:
question: the logic for submitStorageModeForm and submitEncryptModeForm seems mostly repetitive; what was the consideration behind splitting them in commit 1de0e65, vs having an encryptSubmission middleware (original implementation)?

Reply:
Okay, so I went with duplicating the code for commit https://github.com/opengovsg/FormSG/commit/1de0e650c52a419cc8d26e6ed76bef677492ace8 so that I just need to retrieve the form definition once, then 1) perform the relevant checks (submission limits, etc), 2) perform submission validation, 3) encrypt submission payload, then 4) submit the encrypted payload. To be very honest, I think I went with just copying the code then inserting step 2 so that I will get something working first.
I think this isn't the best way of doing this too though, with the amount of code duplication and its not very maintainable. Maybe we can refactor to do something like the [ensure middlewares suggested in Ken's doc](https://www.notion.so/opengov/API-overview-ensure-middlewares-e15206c1f7994e4ebd64d0ca06d79acc)?

Reply:
Cool, thanks @LinHuiqing !

Reply:
Done from [132021e](https://github.com/opengovsg/FormSG/pull/6595/commits/132021ee7a92d805a4482dd8a8ec8073b243ddb9) onwards!